### PR TITLE
Commit offset on consumer close

### PR DESCRIPTION
**Pull request by pierDipi**
When a consumer is closed, we want to commit the offsets for all
owned partitions.

Fixes (some flaky tests)

## Proposed Changes

- Commit offset on consumer close

**Release Note**

```release-note
Commit offset on consumer close.
```

**Docs**

```
None
```

### CODE REVIEW

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/AsyncCloseable.java [hunk 1]
```diff
@@ -61,7 +62,9 @@ static AutoCloseable toAutoCloseable(AsyncCloseable closeable) {
   static AsyncCloseable compose(AsyncCloseable... closeables) {
     return () -> CompositeFuture.all(
       Arrays.stream(closeables)
+        .filter(Objects::nonNull)
         .map(AsyncCloseable::close)
+        .filter(Objects::nonNull)
```
Comment (matzew) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/AsyncCloseable.java:67]: should we reflect a bit what we actually do here w/ the streams APIs, on the javadoc of the `compose` method? It's currently pretty vague

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OffsetManager.java [hunk 2]
```diff
@@ -133,6 +136,22 @@ private synchronized void commit(final TopicPartition topicPartition, final Offs
...
+      this.offsetTrackers.entrySet()
+        .stream()
+        .map(e -> commit(e.getKey(), e.getValue()))
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList())
+    ).mapEmpty();
+  }
+
+  @Override
+  public Future<Void> close() {
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OffsetManager.java:153]: ah, autoclosebable FTW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OffsetManager.java [hunk 3]
```diff
@@ -133,6 +136,22 @@ private synchronized void commit(final TopicPartition topicPartition, final Offs
         .onFailure(cause -> logger.error("failed to commit topic partition {} offset {}", topicPartition, newOffset, cause))
         .mapEmpty();
     }
+    return null;
+  }
+
+  private Future<Void> commitAll() {
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OffsetManager.java:142]: little bit of doc?

**Review by matzew — CHANGES_REQUESTED** [review]
nit - a bit of comments

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by matzew** [comment]
> @pierDipi: #1463 failed to apply on top of branch "release-0.26":

So... it be ? :man_shrugging:

**Comment by pierDipi** [comment]
I'll do it manually :smile:

**Pull request merged**